messaging-client-databus/src/main/java/com/inmobi/databus/partition/CollectorReader.java

```diff
@@ -262,4 +262,18 @@ public MessageCheckpoint getMessageCheckpoint() {
   public MessageCheckpoint buildStartPartitionCheckpoints() {
     return null;
   }
+
+  @Override
+  public Long getReaderBackLog() throws IOException {
+    Long pendingSize = 0l;
+    if (reader == cReader) {
```

Comment:
We should consider pending size from both collector stream(/data) & local stream (/stream_local).

**total pending size = pending data in collector stream + pending data in local stream**